chardet/charsetprober.py

```diff
@@ -79,8 +78,8 @@ def filter_international_words(buf):
 
         This filter applies to all scripts which do not use English characters.
         """
-        filtered = BytesIO()
 
+        out = b''
```

Comment:
I agree supporting both Python versions is difficult, but I'm not quite willing to leave Python 2 users completely in the dust yet, since there are so many of them.  Especially when the hard work for maintaining compatibility has mostly been done already.

That said, I'll definitely target Python 3 for optimizations.